lib/new-settings-loader.js

```diff
@@ -4,14 +4,34 @@ const migration = require('./config-migration')
 
 const OLD_SETTINGS_LOADER_SCHEMA_VERSION = 1
 const NEW_SETTINGS_LOADER_SCHEMA_VERSION = 2
+const PASSWORD_FILLED = 'PASSWORD_FILLED'
+const SECRET_FIELDS = [
+  'bitgo.BTCWalletPassphrase',
+  'bitgo.LTCWalletPassphrase',
+  'bitgo.ZECWalletPassphrase',
+  'bitgo.BCHWalletPassphrase',
+  'bitgo.DASHWalletPassphrase',
+  'bitstamp.secret',
+  'infura.apiSecret',
+  'itbit.clientSecret',
+  'kraken.privateKey',
+  'twilio.authToken'
+]
 
 const accountsSql = `update user_config set data = $2, valid = $3, schema_version = $4 where type = $1;
 insert into user_config (type, data, valid, schema_version)
 select $1, $2, $3, $4 where $1 not in (select type from user_config)`
-function saveAccounts (accountsToSave) {
+function saveAccounts (accounts) {
   return loadAccounts()
     .then(currentAccounts => {
-      const newAccounts = _.assign(currentAccounts, accountsToSave)
+      const serviceCodes = _.keys(accounts)
+      const filledAccountInfo = _.map(code => { return { [code]: _.omitBy(_.isEmpty, accounts[code]) } }, serviceCodes)
+      const accountsToSave = _.reduce((result, item) => {
+        var key = Object.keys(item)[0]
+        result[key] = item[key]
+        return result
+      }, {}, filledAccountInfo)
```

Comment:
This code is pretty hard to understand. What's the goal here?

Reply:
Yeah, I agree... Transforms a list of objects into a single object.